easybuild/tools/systemtools.py

```diff
@@ -574,6 +574,33 @@ def get_cpu_features():
     return cpu_feat
 
 
+def get_gpu_info():
+    """
+    Get the GPU info
+    """
+    gpu_info = {}
+    os_type = get_os_type()
+
+    if os_type == LINUX:
+        try:
+            cmd = "nvidia-smi --query-gpu=gpu_name,driver_version --format=csv,noheader"
+            _log.debug("Trying to determine NVIDIA GPU info on Linux via cmd '%s'", cmd)
+            out, ec = run_cmd(cmd, force_in_dry_run=True, trace=False, stream_output=False)
+            if ec == 0:
+                for line in out.strip().split('\n'):
+                    if 'NVIDIA' in gpu_info and line in gpu_info['NVIDIA']:
+                        gpu_info['NVIDIA'][line] += 1
+                    else:
+                        gpu_info['NVIDIA'] = {}
+                        gpu_info['NVIDIA'][line] = 1
```

Comment:
`setdefault` is your friend here, no need for the `if`/`else`:

```python
for line in out.strip().split('\n'):
    nvidia_gpu_info = gpu_info.setdefault('NVIDIA', {})
    nvidia_gpu_info.setdefault(line, 0)
    nvidia_gpu_info[line] += 1
```

Reply:
Thanks. Done in 2e34420